core-ui/src/components/Extensibility/helpers/jsonataWrapper.js

```diff
@@ -14,6 +14,16 @@ export function jsonataWrapper(expression) {
     );
   });
 
+  exp.registerFunction(
+    'matchByResource',
+    (resource, resourceKind, resourceName) => {
+      return (
+        resource?.involvedObject?.name === resourceName &&
+        resource?.involvedObject?.kind === resourceKind
+      );
+    },
+  );
```

Comment:
For me it's too specific to name this function `matchByResource`, while it's strongly tied to `Events` - only those use `involvedObject` with `kind` and `name`.

Reply:
I changed it to `matchEvents`.